File: Lua_5.4/1024 - Criptografia.lua
--- author: Josélio de S. C. Júnior <joseliojrx25@gmail.com>
--- copyright: Josélio de S. C. Júnior 2021

local a = {}
for _ = 1, io.read(), 1 do

    local input = io.read()

    local function s(x)
        local t = {}
        for i in string.gmatch(x, '.') do table.insert(t, i) end
        return t
    end

    local s1 = ''
    for _, e in ipairs(s(input)) do
        if string.match(e, '%a') then
            s1 = s1..string.char(utf8.codepoint(e) + 3)
        else
            s1 = s1..e
        end
    end

    local s2 = s(string.reverse(s1))
    local s3 = ''

    local m = #s2 / 2
    for i = 1, m , 1 do s3 = s3..s2[i] end
    if #s2 % 2 == 0 then
        for i = m + 1, #s2, 1 do
            s3 = s3..string.char(utf8.codepoint(s2[i]) - 1)
        end
    else
        for i = m + 0.5, #s2, 1 do
            s3 = s3..string.char(utf8.codepoint(s2[i]) - 1)
        end
    end
    table.insert(a, s3)
end

for _, e in ipairs(a) do print(e) end

File: JavaScript/1024 - Criptografia.js
/**
 * @author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
 * @copyright Josélio Júnior (Lunatic Fox) 2021
*/

const p = require('fs').readFileSync('/dev/stdin', 'utf8');
const l = p.split('\n');

const arr = Array.from(Array(+l.shift()), _ => {
    const newArr = l.shift().split('')
        .map(e => e.match(/[A-Z_a-z]/) ? String.fromCharCode(e.charCodeAt() + 3) : e)
        .reverse();

    return [...newArr.slice(0, newArr.length / 2),
        ...newArr.slice(newArr.length / 2, newArr.length)
        .map(e => String.fromCharCode(e.charCodeAt() - 1))
    ].join('');
}).join('\n');

console.log(arr);